clients/src/main/java/org/apache/kafka/clients/consumer/internals/PartitionAssignorAdapter.java

```diff
@@ -0,0 +1,125 @@
+    }
+
+    @Override
+    public void onAssignment(Assignment assignment) {
+        oldAssignor.onAssignment(newToOldAssignment(assignment));
+    }
+
+    @Override
+    public void onAssignment(Assignment assignment, int generation) {
+        oldAssignor.onAssignment(newToOldAssignment(assignment), generation);
+    }
+
+    @Override
+    public String name() {
+        return oldAssignor.name();
+    }
+
+    private org.apache.kafka.clients.consumer.internals.PartitionAssignor.Assignment newToOldAssignment(Assignment assignment) {
+        return new org.apache.kafka.clients.consumer.internals.PartitionAssignor.Assignment(
+            assignment.consumerData().partitions(),assignment.userData());
+    }
+
+    private Map<String, org.apache.kafka.clients.consumer.internals.PartitionAssignor.Subscription> newToOldGroupSubscription(GroupSubscription subscriptions) {
+        Map<String, org.apache.kafka.clients.consumer.internals.PartitionAssignor.Subscription> oldSubscriptions = new HashMap<>();
+        for (Map.Entry<String, Subscription> entry : subscriptions.groupSubscription().entrySet()) {
+            String member = entry.getKey();
+            Subscription newSubscription = entry.getValue();
+            oldSubscriptions.put(member, new org.apache.kafka.clients.consumer.internals.PartitionAssignor.Subscription(
+                        newSubscription.consumerData().topics(), newSubscription.userData()));
+        }
+        return oldSubscriptions;
+    }
+
+    private GroupAssignment oldToNewGroupAssignment(Map<String, org.apache.kafka.clients.consumer.internals.PartitionAssignor.Assignment> assignments) {
+        Map<String, Assignment> newAssignments = new HashMap<>();
+        for (Map.Entry<String, org.apache.kafka.clients.consumer.internals.PartitionAssignor.Assignment> entry : assignments.entrySet()) {
+            String member = entry.getKey();
+            org.apache.kafka.clients.consumer.internals.PartitionAssignor.Assignment oldAssignment = entry.getValue();
+            newAssignments.put(member, new Assignment(new ConsumerAssignmentData(oldAssignment.partitions()), oldAssignment.userData()));
+        }
+        return new GroupAssignment(newAssignments);
+    }
+
+    public static List<PartitionAssignor> getAssignorInstances(AbstractConfig config, String key) {
+        List<PartitionAssignor> assignorInstances = new ArrayList<>();
+        Class<PartitionAssignor> newClass = PartitionAssignor.class;
+        Class<org.apache.kafka.clients.consumer.internals.PartitionAssignor> oldClass = org.apache.kafka.clients.consumer.internals.PartitionAssignor.class;
+
+        List<String> classNames = config.getList(key);
+        if (classNames == null)
+            return assignorInstances;
+
+        for (Object klass : classNames) {
+            Object assignor;
+            if (klass instanceof String) {
+                try {
+                    assignor = Utils.newInstance((String) klass, newClass);
```

Comment:
For existing implementors that rely on `oldClass` but the config map only give the class name string, now it will still break right?